tools/ci_build/github/apple/default_full_apple_framework_build_settings.json

```diff
@@ -0,0 +1,50 @@
+{
+    "build_osx_archs": {
+        "iphoneos": [
+            "arm64"
+        ],
+        "iphonesimulator": [
+            "arm64",
+            "x86_64"
+        ],
+        "macosx": [
+            "arm64",
+            "x86_64"
+        ]
+    },
+    "build_params": {
+        "macosx": [
+            "--macosx",
+            "--parallel",
+            "--use_xcode",
+            "--build_apple_framework",
+            "--use_coreml",
+            "--use_xnnpack",
+            "--skip_tests",
+            "--cmake_extra_defines=onnxruntime_BUILD_UNIT_TESTS=OFF",
+            "--apple_deploy_target=10.14"
+        ],
+        "iphoneos": [
+            "--ios",
+            "--parallel",
+            "--use_xcode",
+            "--build_apple_framework",
+            "--use_coreml",
+            "--use_xnnpack",
+            "--skip_tests",
+            "--cmake_extra_defines=onnxruntime_BUILD_UNIT_TESTS=OFF",
+            "--apple_deploy_target=12.0"
+        ],
+        "iphonesimulator": [
+            "--ios",
+            "--parallel",
```

Comment:
hm, there's quite a lot of duplication of build options with this setup which means more places to update. is it possible to reduce that?
one idea is to have an additional list of common build options that apply to all builds. and then the specific ones, e.g. for iOS, could be limited to `--ios` and the iOS `--apple_deploy_target`.